ast.ts

```diff
@@ -37,6 +37,8 @@ export type Expr<A> =
   | {  a?: A, tag: "lookup", obj: Expr<A>, field: string }
   | {  a?: A, tag: "method-call", obj: Expr<A>, method: string, arguments: Array<Expr<A>> }
   | {  a?: A, tag: "construct", name: string }
+  | {  a?: A, tag: "dict", value: Map<Expr<A>, Expr<A>> }
+  | {  a?: A, tag: "key-lookup", name:string, key:Expr<A> }
```

Comment:
Should `name` be an expression? What about `d[k][k2]`?

Reply:
Fixed this and it's now bracket-lookup like discussed [here](https://github.com/ucsd-cse231-w21/chocopy-wasm-compiler/pull/9#discussion_r583870506)